converter/diag/diagnostics.go

```diff
@@ -62,3 +62,16 @@ func (ds *Diagnostics) Add(severity Severity, message string) {
 		Message:  message,
 	})
 }
+
+// Error implements error.
+func (ds Diagnostics) Error() string {
+	var errorMessage string
+	for ix, diag := range ds {
+		errorMessage += diag.Error()
+		if ix+1 < len(ds) {
+			errorMessage += "\n"
+		}
+	}
+
+	return errorMessage
```

Comment:
nit: use `strings.Builder` instead, which is more efficient and also handles newlines for you:

```suggestion
	var sb strings.Builder
	for ix, diag := range ds {
		fmt.Fprint(&sb, diag.Error())
		if ix+1 < len(ds) {
			fmt.Fprintln(&sb) 
		}
	}

	return sb.String()
```